jwt/src/main/java/me/uport/sdk/jwt/JWTTools.kt

```diff
@@ -295,27 +312,50 @@ class JWTTools(
             PublicKey(pkBytes.toBigInteger()).normalize()
 
         }.filter { publicKey ->
+            try {
+                ecVerify(messageHash, sigData, publicKey)
+            } catch (ex: IllegalArgumentException) {
+                false
+            }
+        }
 
-            ecVerify(messageHash, sigData, publicKey)
+        fun hasEthereumAddressKeys(publicKeys: List<PublicKeyEntry>): Boolean {
+            return publicKeys.any { it.ethereumAddress != null }
+        }
+
+        if (matches.isEmpty() && hasEthereumAddressKeys(publicKeys)) {
+            return verifyRecoverableES256K(publicKeys, signatureBytes, signingInputBytes)
+        } else {
+            return matches.isNotEmpty()
         }
 
-        return matches.isNotEmpty()
     }
 
     private fun verifyRecoverableES256K(
         publicKeys: List<PublicKeyEntry>,
-        sigData: SignatureData,
+        signatureBytes: ByteArray,
         signingInputBytes: ByteArray
     ): Boolean {
 
-        val recoveredPubKey: BigInteger = try {
-            signedJwtToKey(signingInputBytes, sigData)
-        } catch (e: SignatureException) {
-            BigInteger.ZERO
+        val signatures = if (signatureBytes.size == SIG_RECOVERABLE_SIZE) {
+            listOf(signatureBytes.decodeJose())
+        } else {
+            listOf(
+                signatureBytes.decodeJose(0),
```

Comment:
attempting recovery of both variants of the key in case the recovery byte is not present.